app/uk/gov/hmrc/helptosavefrontend/connectors/EmailVerificationConnector.scala

```diff
@@ -0,0 +1,103 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavefrontend.connectors
+
+import com.google.inject.{ImplementedBy, Inject, Singleton}
+import org.joda.time.Period
+import play.api.Configuration
+import play.api.http.Status._
+import play.api.libs.json.{JsValue, Json}
+import uk.gov.hmrc.helptosavefrontend.models.VerifyEmailStatus._
+import uk.gov.hmrc.helptosavefrontend.models.{EmailVerificationRequest, VerifyEmailStatus}
+import uk.gov.hmrc.helptosavefrontend.util.Logging
+import uk.gov.hmrc.play.config.ServicesConfig
+import uk.gov.hmrc.play.http.{HeaderCarrier, HttpResponse}
+import uk.gov.hmrc.play.http.ws.WSHttp
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+@ImplementedBy(classOf[EmailVerificationConnectorImpl])
+trait EmailVerificationConnector {
+
+  def verifyEmail(nino: String, newEmail: String)(implicit hc: HeaderCarrier): Future[VerifyEmailStatus]
+  def isVerified(email: String)(implicit hc: HeaderCarrier): Future[Either[VerifyEmailStatus, Boolean]]
+}
+
+@Singleton
+class EmailVerificationConnectorImpl @Inject() (http: WSHttp, conf: Configuration) extends EmailVerificationConnector with ServicesConfig with Logging {
+
+  val emailVerifyBaseURL = baseUrl("email-verification")
+  val verifyEmailURL = s"$emailVerifyBaseURL/email-verification/verification-requests"
+  def isVerifiedURL(email: String) = s"$emailVerifyBaseURL/email-verification/verified-email-addresses/$email"
+
+  val continueURL = "email-verification.continue.baseUrl"
+  //+ EmailVerificationController.showSuccess.
+  val defaultTemplate = "hts_email_verification"
+
+  override def verifyEmail(nino: String, newEmail: String)(implicit hc: HeaderCarrier): Future[VerifyEmailStatus] = {
+    conf.getInt("services.email-verification.linkTTLMinutes") match {
+      case Some(linkTTLMinutes) ⇒ verify(nino, newEmail, Period.minutes(linkTTLMinutes))
+      case None ⇒ throw new RuntimeException("email-verification.link.ttlMinutes not present in configuration")
+    }
+  }
+
+  private def verify(nino: String, newEmail: String, ttlPeriod: Period)(implicit hc: HeaderCarrier): Future[VerifyEmailStatus] = {
+    val postURL = s"""$verifyEmailURL"""
+    val json: JsValue = Json.toJson(EmailVerificationRequest(newEmail, nino, "", ttlPeriod.toString, ttlPeriod.toString, Map()))
+
+    http.POST(postURL, json).map { (response: HttpResponse) ⇒
+      response.status match {
+        case OK | CREATED =>
+          logger.info("[EmailVerification] - Successful return of data")
+          Verifing(nino, newEmail)
+        case BAD_REQUEST ⇒
+          logger.warn("[EmailVerification] - Bad Request")
```

Comment:
you'll never get a this status or a 500 in this block because the `POST` method on `http` will throw - use  `post` from `uk.gov.hmrc.helptosavefrontend.config.WSHttpExtension` instead to be able to handle these